visearch-android/src/main/java/com/visenze/visearch/android/ProductBaseSearchParams.java

```diff
@@ -0,0 +1,470 @@
+    }
+
+    public String getVaAppBundleId() {
+        return vaAppBundleId;
+    }
+
+    public void setVaAppBundleId(String vaAppBundleId) {
+        this.vaAppBundleId = vaAppBundleId;
+    }
+
+    public String getVaAppName() {
+        return vaAppName;
+    }
+
+    public void setVaAppName(String vaAppName) {
+        this.vaAppName = vaAppName;
+    }
+
+    public String getVaAppVersion() {
+        return vaAppVersion;
+    }
+
+    public void setVaAppVersion(String vaAppVersion) {
+        this.vaAppVersion = vaAppVersion;
+    }
+
+    public String getVaAaid() {
+        return vaAaid;
+    }
+
+    public void setVaAaid(String vaAaid) {
+        this.vaAaid = vaAaid;
+    }
+
+    public String getVaDidmd5() {
+        return vaDidmd5;
+    }
+
+    public void setVaDidmd5(String vaDidmd5) {
+        this.vaDidmd5 = vaDidmd5;
+    }
+
+    public String getVaN1() {
+        return vaN1;
+    }
+
+    public void setVaN1(String vaN1) {
+        this.vaN1 = vaN1;
+    }
+
+    public String getVaN2() {
+        return vaN2;
+    }
+
+    public void setVaN2(String vaN2) {
+        this.vaN2 = vaN2;
+    }
+
+    public String getVaS1() {
+        return vaS1;
+    }
+
+    public void setVaS1(String vaS1) {
+        this.vaS1 = vaS1;
+    }
+
+    public String getVaS2() {
+        return vaS2;
+    }
+
+    public void setVaS2(String vaS2) {
+        this.vaS2 = vaS2;
+    }
+
+
+    @SerializedName("filters")
+    private List<String> filters;
+
+    @SerializedName("text_filters")
+    private List<String> textFilters;
+
+    @SerializedName("attrs_to_get")
+    private List<String> attrsToGet;
+
+    @SerializedName("facets")
+    private List<String> facets;
+
+
+    @SerializedName("custom_map")
+    @Expose(deserialize = false, serialize = false)
+    private Map<String, String> customMap;
+
+
+
+    public List<String> getFilters() {
+        return filters;
+    }
+
+    public void setFilters(List<String> filters) {
+        this.filters = filters;
+    }
+
+    public List<String> getTextFilters() {
+        return textFilters;
+    }
+
+    public void setTextFilters(List<String> textFilters) {
+        this.textFilters = textFilters;
+    }
+
+    public List<String> getAttrsToGet() {
+        return attrsToGet;
+    }
+
+    public void setAttrsToGet(List<String> attrsToGet) {
+        this.attrsToGet = attrsToGet;
+    }
+
+    public List<String> getFacets() {
+        return facets;
+    }
+
+    public void setFacets(List<String> facets) {
+        this.facets = facets;
+    }
+
+    public Map<String, String> getCustomMap() {
+        return customMap;
+    }
+
+    public void setCustomMap(Map<String, String> customMap) {
+        this.customMap = customMap;
+    }
+
+    public RetrofitQueryMap getQueryMap() {
+
+        Map<String, Object> map = new HashMap<String, Object>();
+        RetrofitQueryMap ret = new RetrofitQueryMap(map);
+
+        Gson gson = new Gson();
```

Comment:
can this object be made "static" for re-use across multiple classes? or do we have to create a new object always?

Reply:
changed already